app/assets/v2/js/pages/profile-trust.js

```diff
@@ -657,6 +657,152 @@ Vue.component('sms-verify-modal', {
+      validated: false,
+      verificationEthAddress: '',
+    };
+  },
+  mounted: function() {
+    $(document).on('click', '#verify-ens-link', function(event) {
+      event.preventDefault();
+      this.showValidation = true;
+    }.bind(this));
+    this.checkENSValidation();
+  },
+  template: `
+    <b-modal id="ens-modal" @hide="dismissVerification()" :visible="showValidation" center hide-header hide-footer>
+      <template v-slot:default="{ hide }">
+        <div class="mx-5 mt-5 mb-4 text-center">
+          <div class="mb-3">
+            <h1 class="font-bigger-4 font-weight-bold">Verify with ENS</h1>
+            <p>
+              ENS is a name service built on Ethereum. It offers a secure and decentralized way to address resources using human-readable names.
+              <p class="mb-4">
+                <a href="https://ens.domains/about">Learn more.</a>
+              </p>
+            </p>
+          <div>
+            <h2 class="font-bigger-4 font-weight-bold">You need to have these requirements:</h2>
+            <ol>
+              <li>Setup your address. <span :class="\`mr-2 fas fa-\${ validationStep > 1 ? 'check check gc-text-green' : 'times gc-text-pink' }\`"></span>
+                <input v-model="verificationEthAddress" class="form-control" type="text" @onchange="testVerification()">
+                <a href="#" @click.prevent.stop="pullEthAddress()">Pull address from Metamask</a><br>
+              </li>
+              <li>Your address has an ENS domain associated, you can get one <a href="https://app.ens.domains/">here</a>. <span :class="\`mr-2 fas fa-\${ validationStep > 2 ? 'check gc-text-green' : 'times gc-text-pink' }\`"></span></li>
+              <li>Your address should match with the address associated to the ENS domain. <span :class="\`mr-2 fas fa-\${ validationStep > 5 ? 'check gc-text-green' : 'times gc-text-pink' }\`"></span> </li>
+            </ol>
+          </div>
+        </div>
+        <div>
+          <p class="mb-4" v-if="!validated && validationError === false">
+            You fulfill all the requirements, just click on validate to confirm that your ENS account is valid
+          </p>
+          <b-button @click="verifyENS" class="btn-gc-blue mt-3 mb-2" size="lg" v-if="!validated && validationError === false">
+            Validate
+          </b-button>
+          <div v-if="validationError & validationErrorMsg !== ''" style="color: red">
+            <small>[[validationErrorMsg]]</small>
+          </div>
+          <br />
+          <a href="" v-if="validationError" @click.f="dismissVerification()">
+            Go Back
+          </a>
+        </div>
+        <div v-if="validated && !validationError">
+          Your ENS verification was successful. Thank you for helping make Gitcoin more sybil resistant!
+          <a href="" class="btn btn-gc-blue px-5 mt-3 mb-2 mx-2" role="button" style="font-size: 1.3em">Done</a>
+        </div>
+      </div>
+    </template>
+  </b-modal>`,
```

Comment:
For neater code, i think it'd make more sense for the template to be in a seperate file. Viewing HTML as a large string kinda makes it harder to read.

Reply:
i extracted the template into new file